pilot/pkg/bootstrap/server.go

```diff
@@ -184,10 +223,24 @@ func NewServer(args PilotArgs) (*Server, error) {
 	if err := s.initClusterRegistries(&args); err != nil {
 		return nil, fmt.Errorf("cluster registries: %v", err)
 	}
+
+	// Will run the sidecar injector in pilot - this precedes istiod, which does the same thing but with different
+	// config - certs are local. The code in main() hardcodes port 15017 and InjectionDirectory /etc/istio/inject
+	// and only operates if the directory exists - which is not set in either old or new installer. This code
+	// was used before sidecar injector was split to a separate deployment. Instead of attempting to use the old
+	// code, we'll use the istiod code which was tested more recently.
```

Comment:
This is not old code, I just added this a few days ago?

Reply:
Ah, missed that PR. We used to have it in pilot in the early days, I assumed it was left over.

I made few changes to the implementation ( to patch )